lib/puppet/type/cloudwatch_alarm.rb

```diff
@@ -0,0 +1,91 @@
+Puppet::Type.newtype(:cloudwatch_alarm) do
+  @doc = 'type representing an AWS CloudWatch Alarm'
+
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'the name of the alarm'
+    validate do |value|
+      fail 'alarms must have a name' if value == ''
+    end
+  end
+
+  newproperty(:metric) do
+    desc 'the name of the metric to track'
+    validate do |value|
+      fail 'metric must not be blank' if value == ''
+    end
+  end
+
+  newproperty(:namespace) do
+    desc 'the namespace of the metric to track'
+    validate do |value|
+      fail 'namespace must not be blank' if value == ''
+    end
+  end
+
+  newproperty(:statistic) do
+    desc 'the statistic to track for the metric'
+    validate do |value|
+      fail 'statistic must not be blank' if value == ''
+    end
+  end
+
+  newproperty(:period) do
+    desc 'the periodicity of the alarm check'
+    validate do |value|
+      fail 'period cannot be blank' if value == ''
+    end
+    def insync?(is)
```

Comment:
Why override insync? instead of munging? Does the `is` value typically return a string?